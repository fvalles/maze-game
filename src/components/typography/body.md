Basic body text component with default color="text":

```jsx
<Body>Lorem ipsum</Body>
```

Body text component with color="white":

```jsx
<Body color="white">Lorem ipsum</Body>
```
