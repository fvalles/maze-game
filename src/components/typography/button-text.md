Basic button text component with default color="white":

```jsx
<ButtonSmall>Lorem ipsum</ButtonSmall>
```

Button text component with color="text":

```jsx
<ButtonSmall color="text">Lorem ipsum</ButtonSmall>
```
