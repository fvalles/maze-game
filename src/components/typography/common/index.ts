import { KeyColors } from '../../../core/theme/colors';

/**
 * Types
 */

export interface TypographyProps {
  /** Text color */
  color?: KeyColors;
}
