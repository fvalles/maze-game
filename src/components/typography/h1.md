Basic h1 text component with default color="text":

```jsx
<H1>Lorem ipsum</H1>
```

H1 text component with color="white":

```jsx
<H1 color="white">Lorem ipsum</H1>
```
