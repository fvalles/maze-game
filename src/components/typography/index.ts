export { Body } from './body';
export { ButtonText } from './button-text';
export { H1 } from './h1';
export { H2 } from './h2';
