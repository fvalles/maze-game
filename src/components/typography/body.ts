import styled from 'styled-components';
import { TypographyProps } from './common';

export const Body = styled.p<TypographyProps>`
  color: ${({ color = 'text', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.light}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 20px;
  line-height: 24px;
`;
