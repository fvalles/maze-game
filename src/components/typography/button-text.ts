import styled from 'styled-components';
import { TypographyProps } from './common';

export const ButtonText = styled.span<TypographyProps>`
  color: ${({ color = 'white', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 20px;
  font-weight: bold;
  line-height: 16px;
`;
