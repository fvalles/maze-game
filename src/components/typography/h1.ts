import styled from 'styled-components';
import { TypographyProps } from './common';

export const H1 = styled.h1<TypographyProps>`
  color: ${({ color = 'text', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 58px;
  line-height: 62px;
`;
