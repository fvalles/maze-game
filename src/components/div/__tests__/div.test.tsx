import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { H1 } from '@components/typography';
import { Div } from '..';

describe('div component', () => {
  it('should render with default css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Div>
            <H1>Some text</H1>
          </Div>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('width', '100%');
    expect(result).toMatchSnapshot();
  });

  it('should render with custom css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Div
            alignItems="center"
            boxShadow="3px 3px"
            display="flex"
            flexDirection="column"
            justifyContent="center"
            margin="16px"
            padding="10px">
            <H1>Some text</H1>
          </Div>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('align-items', 'center');
    expect(result).toHaveStyleRule('box-shadow', '3px 3px');
    expect(result).toHaveStyleRule('display', 'flex');
    expect(result).toHaveStyleRule('flex-direction', 'column');
    expect(result).toHaveStyleRule('justify-content', 'center');
    expect(result).toHaveStyleRule('margin', '16px');
    expect(result).toHaveStyleRule('padding', '10px');
    expect(result).toHaveStyleRule('width', '100%');
    expect(result).toMatchSnapshot();
  });
});
