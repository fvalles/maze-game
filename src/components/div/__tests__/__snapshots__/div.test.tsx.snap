// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`div component should render with custom css properties 1`] = `
.c1 {
  color: #212240;
  font-family: Avenir Regular,'Avenir Next','Segoe UI', sans-serif;
  font-size: 58px;
  line-height: 62px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 3px 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 16px;
  padding: 10px;
  width: 100%;
}

<div
  className="c0"
  display="flex"
>
  <h1
    className="c1"
  >
    Some text
  </h1>
</div>
`;

exports[`div component should render with default css properties 1`] = `
.c1 {
  color: #212240;
  font-family: Avenir Regular,'Avenir Next','Segoe UI', sans-serif;
  font-size: 58px;
  line-height: 62px;
}

.c0 {
  width: 100%;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Some text
  </h1>
</div>
`;
